### ISSUE TITLE
Since @grpc/grpc-js `v1.9.0`,  channel is stuck in `CONNECTING` state for ~20s when reconnecting

### ISSUE THREAD

**Issue by toddbaert** [grpc/grpc-node#2620]
### Problem description

Reconnect is slowed in @grpc/grpc-js `v.1.9.0+`. Initial connection is instant, but if a connection is lost, the connection gets "stuck" in `CONNECTING` state.

### Reproduction steps

- connect to a server using a `Client` without TLS and without any particular client options:
```ts
new MyClient(
  'localhost:8888',
  credentials.createInsecure(),
);
```
- add an interval, with a statement like somewhere in your client code:
```ts
    setInterval(() => {
      console.log(this._client.getChannel().getConnectivityState(true)); // attempt to get the channel state and connect.
    }, 1000);
```
- restart the server (in this case at 'localhost:8888`)
- observe that long after the server is up and able to accept requests, `1` (meaning `ConnectivityState.CONNECTING`) is printed to the console every 1s (indicating the service is connecting, but not yet connected).
- wait ~20 seconds
- observe that `2` (meaning `ConnectivityState.READY`) in printed to the console every 1s, and your client is now usable.

This behavior is new since version `1.9.x`. In prior versions, the reconnection would be nearly instant (as soon as the server is up, the client is able to connect, and `2` is printed).

### Environment
 - Linux 6.5.9-arch2-1 x86_64 (:warning: please note, my colleague was unable to reproduce the issue on their macOS system)
 - Node version v20.9.0
 - Using nvm
 - @grpc/grpc-js@1.9.xx

### Additional context

- this DOES NOT impact the initial connection. The initial connection remains as immediate as in `1.8.xx` versions.
- I tried tweaking some of the options, such as `grpc.keepalive_time_ms` and `grpc.max_reconnect_backoff_ms`, but they seemed to have no impact on the issue
- the 20 second delay is VERY consistent. The connection is never ready before 20s, and always ready before 25s. It seems like some timer is blocking the connection readiness.

 I looked through the changes mentioned in the 1.9.0 release notes and can't see anything suspicious as to the cause. I noticed [this change](https://github.com/grpc/grpc-node/pull/2471), but I don't believe this has anything to do idleness.

As a workaround, we've pinned our version to `~1.8.x` for now.

**Comment by toddbaert** [comment]
> This looks like the bug that was fixed in #2619. Please try updating to version 1.9.11.

I will try this today!

**Comment by toddbaert** [comment]
@murgatroid99 I can confirm this is resolved with `1.9.11`. Thanks for your time and attention!
